### PR TITLE
Navigation Bar

### PR DESCRIPTION
**Pull request by khaledsaadeh**
Added the navigation bar with search functionality, search has a minor bug that will be handled in a later PR

### CODE REVIEW

File: src/components/NavBar.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+const NavBar = (props) => {
+  const API =
+    "https://api.themoviedb.org/3/search/movie?api_key=4f298a53e552283bee957836a529baec&language=en-US&page=1&include_adult=false&query=";
```
Comment (hazemturki) [inline src/components/NavBar.js:5]: Not sure this is the best place for such a constant.
Doesn't seem appropriate, this is a NavBar component it shouldn't define a API URL

File: src/components/NavBar.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const [search, setSearch] = useState("");
+  
+  const [movies, setMovies] = useState([]);  
+  const searchHandler = (event) => {
+    event.preventDefault();
+    if (search.trim() !== "") {
+      axios.get(API + search).then((res) => {
```
Comment (hazemturki) [inline src/components/NavBar.js:12]: Please don't use simple string concatenation to add the query and instead use Axios.get options.
Please refer to the official axios docs

```suggestion
      axios.get(API, {params: search}).then((res) => {
```

File: src/components/NavBar.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+const NavBar = (props) => {
+  const API =
+    "https://api.themoviedb.org/3/search/movie?api_key=4f298a53e552283bee957836a529baec&language=en-US&page=1&include_adult=false&query=";
+  const [search, setSearch] = useState("");
+  
+  const [movies, setMovies] = useState([]);  
+  const searchHandler = (event) => {
```
Comment (hazemturki) [inline src/components/NavBar.js:9]: This search handler should trigger a callBackFunction to the app, but it shouldn't be responsible for fetching data.

File: src/App.js [hunk 4]
```diff
@@ -14,10 +15,16 @@ function App() {
...
-  console.log(movies);
+
+  const showSearchResults = (movies) => {
+    setMovies(movies);
+  };
+
   return (
     <div className="App">
-      <MovieList movies={movies}  />
+      <NavBar passToApp={showSearchResults} />
```
Comment (hazemturki) [inline src/App.js:25]: Pass to app is an ambiguous name for a prop.
Please rename this according to the other comments in the review.

```suggestion
      <NavBar ?={showSearchResults} />
```

**Review by hazemturki — CHANGES_REQUESTED** [review]

**Comment by khaledsaadeh** [comment]
-Added API constants to the main component.
-Used the {params} option from axios.get instead of string concatenation.
-Search handler now triggers a function responsible for fetching the data, instead of fetching them itself.
-changed the passToApp prop name to a more appropiate prop name.

**Review by hazemturki — APPROVED** [review]

**Pull request merged**